time.js

```diff
@@ -8,21 +8,321 @@
+const ohItem = Java.type('org.openhab.core.items.Item');
+
+/**
+ * Converts the Java ZonedDateTime to a time.ZonedDateTime
+ * @param {java.time.ZonedDateTime} zdt date time to convert to a time.ZonedDateTime
+ * @returns {time.ZonedDateTime}
+ */
+const javaZDTtoZDT = function(zdt) {
+    const epoch = zdt.toInstant().toEpochMilli();
+    const instant = time.Instant.ofEpochMilli(epoch);
+    const zone = time.ZoneId.of(zdt.getZone().toString());
+    return time.ZonedDateTime.ofInstant(instant, zone);
+}
+
+/**
+ * Adds millis to the passed in ZDT as milliseconds. The millis is rounded 
+ * first. If millis is negative they will be subtracted.
+ * @param {number|bigint} millis number of milliseconds to add
+ */
+const addMillisToNow = function(millis) {
+    return time.ZonedDateTime.now().plus(Math.round(millis), time.ChronoUnit.MILLIS);
+}
+
+/**
+ * Adds the passed in QuantityType<Time> to now.
+ * @param {QuantityType} quantityType an Item's QuantityType
+ * @returns now plus the time length in the quantityType
+ * @throws error when the units for the quantity type are not one of the Time units
+ */
+const addQuantityType = function(quantityType) {
+    const secs = quantityType.toUnit('s');
+    if(secs) {
+        return time.ZonedDateTime.now().plusSeconds(secs.doubleValue());
+    } else {
+        throw Error('Only Time units are supported to convert QuantityTypes to a ZonedDateTime: ' + quantityType.toString());
+    }
+}
+
+/**
+ * Tests the passed in string to see if it conforms to the ISO8601 standard
+ * @param {String} dtStr potential ISO8601 string
+ * @returns {boolean} true if ISO8601 format
+ */
+const isISO8601 = (dtStr) => {
+  var regex = new RegExp(/^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$/);
+  return regex.test(dtStr);    
+}
+
+/**
+ * Tests the string to see if it matches a 24 hour clock time
+ * @param {String} dtStr potential HH:MM String
+ * @returns {boolean} true if it matches HH:MM
+ */
+ const is24Hr = function (dtStr) {
+    var regex = new RegExp(/^(0?[0-9]|1[0-9]|2[0-3])(:[0-5][0-9]){1,2}$/);
+    return regex.test(dtStr);
+}
+
+/**
+ * Tests the string to see if it matches a 12 hour clock time
+ * @param {String} dtStr potential hh:MM aa string
+ * @returns {boolean} true if it matches hh:mm aa 
+ */
+const is12Hr = function (dtStr) {
+    var regex = new RegExp(/^(0?[0-9]|1[0-2])(:[0-5][0-9]){1,2} ?[a|p|A|P]\.?[m|M]\.?$/);
+    return regex.test(dtStr);
+}
+
+/**
+ * Parses the passed in string based on it's format and converted to a ZonedDateTime.
+ * @param {String} str string number to try and parse and convert
+ * @throws Error when the string cannot be parsed
+ */
+const parseString = function(str) {
+
+    // Number
+    if(!isNaN(str)) {
+        return addMillisToNow(str);
+    }
+
+    // ISO8601
+    else if(isISO8601(str)) {
```

Comment:
```suggestion
    if(isISO8601(str)) {
```
since each of these if statements return if matched, there no need for else in any of them

Reply:
Done